### PR TITLE
Move Featured Media UI to Featured Image Metabox

### LINKED ISSUE #698: UI to override featured image display on single posts is not clear
In the modal where this is display it's so tiny and at the bottom of the window that it's easy to miss. We should consider adding it either at the top or maybe in the right column with the photo info so that the context is clear.

We might also want to add some explanatory text to make it clear that this means it will be hidden in the hero area of the single post but still used as the thumbnail on the homepage and elsewhere on the site.

### PR DESCRIPTION
**Pull request by rclations**
fixes #698

## The Problem
- Featured image override checkbox is hidden at the bottom of the featured image selection modal
- The wording on the checkbox is confusing
- Options in the image selection modal should apply to that specific image - this checkbox is applies to the post (not the image itself)
- The existing "Set Featured Media" button is located above the wysiwyg editor. This area is generally used for shortcode buttons to be used within the wysiwyg editor.

## Proposed Solution
The featured media metabox was removed so we could build our own button with additional options for featured video, embed, gallery, etc.

This PR adds a new featured media metabox and moves the "Set Featured Media" button there.

### CODE REVIEW

**Comment by aschweigert** [comment]
@rclations same issue here :/ I'm not able to check out this branch to review because it's on your fork. Can you move that branch over here so I can do some final review/cleanup before merging?

**Comment by rclations** [comment]
@aschweigert can you access it now? I just checked the new "allow edits from maintainers" feature

**Comment by aschweigert** [comment]
ok, this lgtm, a couple to-do items before merging:

- [ ] need to make the featured media templates translatable (currently nothing in there is)
- [ ] consider refactoring the duplicative featured media functions (there's a set for each type of featured media that do nearly the same thing, let's try to see if we can refactor that and make it a bit cleaner)

**Comment by rclations** [comment]
should be all set, but @aschweigert or @benlk, could you give this a quick review before I merge?

**Comment by benlk** [comment]
I'm getting a `featured-media.min.js?ver=4.6.1:1 Uncaught TypeError: Cannot read property 'replace' of undefined` when running with `LARGO_DEBUG` set to False, but not when True.

With True, it works pretty well, but should probably be higher in the stack on the page:

screen shot 2016-09-22 at 4 58 12 pm

screen shot 2016-09-22 at 4 58 21 pm
screen shot 2016-09-22 at 4 58 28 pm
screen shot 2016-09-22 at 4 58 34 pm

**Comment by aschweigert** [comment]
@jackbrighton we'll need docs coverage for this since we're moving peoples' button. Let's make sure that gets done before we ship 0.5.5

File: inc/featured-media.php [hunk 1]
```diff
@@ -157,109 +132,51 @@ function largo_get_featured_image_hero($post = null, $classes = '') {
...
-function largo_get_featured_embed_hero($post = null, $classes = '') {
-
-	$the_post = get_post($post);
-	$featured_media = largo_get_featured_media($the_post->ID);
-
-	if( !in_array($featured_media['type'], array('embed-code', 'video')) ) {
-		return;
+	switch( $featured_media['type'] ) {
+		case 'video':
+			$template_slug = 'video';
```
Comment (benlk) [inline inc/featured-media.php]: `partials/hero-featured-video` is not a file that exists.

This line should use `$template_slug = 'embed'`, to match the old code:

```php
if (in_array($featured_media['type'], array('embed-code', 'video'))) {
	$ret = largo_get_featured_embed_hero($post->ID,$classes);
```

File: inc/featured-media.php [hunk 2]
```diff
@@ -157,109 +132,51 @@ function largo_get_featured_image_hero($post = null, $classes = '') {
 		'the_post' => $the_post
```
Comment (benlk) [inline inc/featured-media.php]: If we're adding `partials/hero-featured-embed.php` to the list of things that get rendered with this context, then `$featured_media` and `$gallery_ids` need to be passed in this array if that's appropriate to the rendering.

**Comment by benlk** [comment]
The code changes in this PR included a lot of things that weren't mentioned in the pull request's description. :frowning_face:

**Pull request merged**